settings.py

```diff
@@ -270,6 +270,8 @@ def dump(self, f: TextIO, level: int = 0) -> None:
             # fetch class to avoid going through getattr
             cls = self.__class__
             type_hints = cls.get_type_hints()
+            if type_hints == {}:
```

Comment:
Any reason why you compare to empty dict instead of checking truthiness?

```suggestion
            if not type_hints:
                # write empty dict for empty Group
```

Reply:
The type hint for get_type_hints says it can only return dicts, so we should be safe.